### ISSUE TITLE
Allow form components to have an "initialValue"

### ISSUE THREAD

**Issue by bakura10** [Shopify/polaris#137]
# Issue summary

Hi,

I'm still new to React so I may miss a better approach, but when dealing with forms in the context of SPA, I often want to provide a default value (usually coming from the Redux state) for the input field, and then have another value mapped to a locale state:

```jsx
<Select value={this.props.subscription.plan.id}/>
```

Here, the value is tied to data coming from my store, but when it changes I want to update a locale state.

For now the best approach I've found is using componentWillReceiveProps to update the state value when the data is loaded from Redux:

```jsx
componentWillReceiveProps(nextProps) {
    const {subscription} = nextProps;
    this.setState({planId: subscription.planId});
  }
```

And render it this way:

```jsx
<Select value={this.state.planId}/>
```

I wonder if instead of the boilerplate, we could have something like that:

```jsx
<Select initialValue={this.props.subscription.plan.id} value={this.state.planId}/>
```

In other words the initialValue will only be used for the initial rendering of the component, while the value is used for subsequent update.

Thanks!

**Comment by lemonmade** [comment]
There's certainly a number of different ways to do this, but I don't think an `initialValue` is the best choice here; it effectively gives you two ways of setting the value, and you could safely ignore one or the other and achieve the same result. The simplest suggestion I have for you is to simply use an `||`; if the state value exists, use it, otherwise fallback on the value passed in via props:

```js
<Select value={this.state.planId || this.props.subscription.plan.id} />
```

In our own form code, we have an abstraction that lets us store the initial value and "current" value separately, which is another direction you could go with this.

Let me know if you don't feel this appropriately addresses your question.
